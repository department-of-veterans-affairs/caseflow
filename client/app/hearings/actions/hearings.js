import * as Constants from '../constants/constants';

export const handleServerError = (err) => ({
  type: Constants.HANDLE_SERVER_ERROR,
  payload: {
    err
  }
});
