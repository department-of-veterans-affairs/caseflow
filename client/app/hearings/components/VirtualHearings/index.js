export * from '../details/HearingEmail';
export * from './Section';
export * from './Timezone';
export * from './HelperText';
