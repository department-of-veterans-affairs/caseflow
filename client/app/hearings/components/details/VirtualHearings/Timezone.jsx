import React from 'react';
import PropTypes from 'prop-types';

import SearchableDropdown from '../../../../components/SearchableDropdown';
import { timezones } from '../../../utils';
import { timezoneDropdownStyles, timezoneStyles } from '../style';

export const Timezone = ({ readOnly, value, errorMessage, onChange, time }) => {
  const { options, commonsCount } = timezones(time);

  return (
    <SearchableDropdown
      styling={timezoneStyles(commonsCount)}
      dropdownStyling={timezoneDropdownStyles(commonsCount)}
      name="Timezone"
      readOnly={readOnly}
      placeholder="Select a timezone"
      options={options}
      value={value}
      onChange={(option) => onChange(option ? option.value : null)}
      errorMessage={errorMessage}
      strongLabel
    />
  );
};

Timezone.defaultProps = {
  value: null,
  readOnly: false
};

Timezone.propTypes = {
  errorMessage: PropTypes.string,
  onChange: PropTypes.func,
  readOnly: PropTypes.bool,
  value: PropTypes.string,
  time: PropTypes.string
};
