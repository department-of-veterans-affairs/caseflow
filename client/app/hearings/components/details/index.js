export * from './Address';
export * from './ReadOnly';
