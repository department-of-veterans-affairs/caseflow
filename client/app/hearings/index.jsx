import React from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import perflogger from 'redux-perf-middleware';
import thunk from 'redux-thunk';

import ConfigUtil from '../util/ConfigUtil';
import DocketsContainer from './DocketsContainer';
import DailyDocketContainer from './DailyDocketContainer';
import HearingWorksheetContainer from './HearingWorksheetContainer';
import { hearingsReducers, mapDataToInitialState } from './reducers/index';
import ScrollToTop from './util/ScrollTop';

const configureStore = (data) => {

  const middleware = [];

  if (!ConfigUtil.test()) {
    middleware.push(thunk, perflogger);
  }

  // This is to be used with the Redux Devtools Chrome extension
  // https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd
  // eslint-disable-next-line no-underscore-dangle
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

  const initialData = mapDataToInitialState(data);
  const store = createStore(
    hearingsReducers,
    initialData,
    composeEnhancers(applyMiddleware(...middleware))
  );

  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('./reducers/index', () => {
      store.replaceReducer(hearingsReducers);
    });
  }

  return store;
};

const Hearings = ({ hearings }) => {

  return <Provider store={configureStore(hearings)}>
    <div>
      <BrowserRouter>
        <div>
         <ScrollToTop />
          <Route exact path="/hearings/dockets"
            component={() => (
              <DocketsContainer
                veteran_law_judge={hearings.veteran_law_judge} />
            )}
          />

          <Route exact path="/hearings/dockets/:date"
            component={(props) => (
              <DailyDocketContainer
                veteran_law_judge={hearings.veteran_law_judge}
                date={props.match.params.date} />
            )}
          />

        <Route exact path="/hearings/:hearingId/worksheet"
            component={(props) => (
              <HearingWorksheetContainer
                veteran_law_judge={hearings.veteran_law_judge}
                hearingId={props.match.params.hearingId} />
            )}
          />
        </div>

      </BrowserRouter>
    </div>
  </Provider>;
};

export default Hearings;
