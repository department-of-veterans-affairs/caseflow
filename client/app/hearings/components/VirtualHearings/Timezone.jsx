import React from 'react';
import PropTypes from 'prop-types';

import SearchableDropdown from '../../../components/SearchableDropdown';
import { timezones } from '../../utils';
import { timezoneStyles } from '../details/style';

export const Timezone = ({
  name,
  label,
  readOnly,
  value,
  errorMessage,
  onChange,
  time,
  roTimezone,
  required,
  optional
}) => {
  const { options, commonsCount } = timezones(time, roTimezone);

  return (
    <SearchableDropdown
      optional={optional}
      required={!readOnly && required}
      styling={timezoneStyles(commonsCount)}
      name={name}
      label={label}
      readOnly={readOnly}
      placeholder="Select a timezone"
      options={options}
      value={value}
      onChange={(option) => onChange(option ? option.value : null)}
      errorMessage={errorMessage}
      strongLabel
    />
  );
};

Timezone.defaultProps = {
  name: 'Timezone',
  value: null,
  readOnly: false
};

Timezone.propTypes = {
  errorMessage: PropTypes.string,
  onChange: PropTypes.func,
  readOnly: PropTypes.bool,
  required: PropTypes.bool,
  optional: PropTypes.bool,
  value: PropTypes.string,
  name: PropTypes.string,
  label: PropTypes.string,
  time: PropTypes.string.isRequired,
  roTimezone: PropTypes.string.isRequired
};
