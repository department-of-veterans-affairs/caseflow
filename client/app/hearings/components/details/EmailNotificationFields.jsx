import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import { marginTop, input8px } from '../details/style';
import { HearingEmail } from './HearingEmail';
import { EmailNotificationHistory } from './EmailNotificationHistory';
import { Timezone } from '../VirtualHearings/Timezone';
import { HelperText } from '../VirtualHearings/HelperText';
import { ContentSection } from '../../../components/ContentSection';
import COPY from '../../../../COPY';
import { getAppellantTitle, readOnlyEmails } from '../../utils';

export const EmailNotificationFields = ({
  errors,
  initialRepresentativeTz,
  hearing,
  readOnly,
  update,
  time,
  roTimezone,
  header
}) => {

  const disableField = readOnly || readOnlyEmails(hearing);
  const appellantTitle = getAppellantTitle(hearing?.appellantIsNotVeteran);

  return (
    <ContentSection header={header}>
      <React.Fragment>
        <div id="email-section" className="usa-grid">
          <div className="usa-width-one-third" {...input8px}>
            <Timezone
              required={hearing?.isVirtual}
              optional={!hearing?.isVirtual}
              errorMessage={errors?.appellantTz}
              value={hearing?.appellantTz}
              onChange={(appellantTz) => update('hearing', { appellantTz })}
              readOnly={disableField}
              time={time}
              roTimezone={roTimezone}
              name="appellantTz"
              label={`${getAppellantTitle(hearing?.appellantIsNotVeteran)} Timezone`}
            />
            <HelperText label={COPY.VIRTUAL_HEARING_TIMEZONE_HELPER_TEXT} />
          </div>
          <div className="usa-width-one-third">
            <HearingEmail
              required={hearing?.isVirtual}
              optional={!hearing?.isVirtual}
              disabled={disableField}
              label={`${appellantTitle} Email`}
              emailType="appellantEmailAddress"
              email={hearing?.appellantEmailAddress}
              error={errors?.appellantEmailAddress}
              update={update}
            />
          </div>
        </div>
        <div className="cf-help-divider" />
        <div className={classNames('usa-grid', { [marginTop(30)]: true })}>
          <div className="usa-width-one-third" {...input8px}>
            <Timezone
              required={Boolean(hearing?.representativeEmailAddress)}
              optional={!hearing?.representativeEmailAddress}
              errorMessage={errors?.representativeTz}
              value={hearing?.representativeTz}
              onChange={(representativeTz) => update('hearing', { representativeTz })}
              readOnly={disableField || !hearing?.representativeEmailAddress}
              time={time}
              roTimezone={roTimezone}
              name="representativeTz"
              label="POA/Representative Timezone"
            />
            <HelperText label={COPY.VIRTUAL_HEARING_TIMEZONE_HELPER_TEXT} />
          </div>
          <div className="usa-width-one-third">
            <HearingEmail
              optional
              disabled={disableField}
              label="POA/Representative Email"
              emailType="representativeEmailAddress"
              email={hearing?.representativeEmailAddress}
              error={errors?.representativeEmailAddress}
              update={(key, value) => {
                // Switch the representative timezone back to the initial value if the
                // representative email is changed to null. This should prevent `deepDiff``
                // from trying to send any changes to the representative timezone if the
                // representative email is being removed.
                if (!value.representativeEmailAddress) {
                  value.representativeTz = initialRepresentativeTz;
                }

                update(key, value);
              }}
            />
          </div>
        </div>
        {hearing?.emailEvents?.length > 0 && (
          <EmailNotificationHistory rows={hearing?.emailEvents} />
        )}
      </React.Fragment>
    </ContentSection>
  );
};

EmailNotificationFields.propTypes = {
  requestType: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
  roTimezone: PropTypes.string.isRequired,
  appellantTitle: PropTypes.string.isRequired,
  readOnly: PropTypes.bool,
  update: PropTypes.func,
  hearing: PropTypes.object,
  errors: PropTypes.object,
  initialRepresentativeTz: PropTypes.string,
  header: PropTypes.string
};
