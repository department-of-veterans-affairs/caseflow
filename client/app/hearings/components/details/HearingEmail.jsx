import PropTypes from 'prop-types';
import React from 'react';
import { isEmpty } from 'lodash';
import classnames from 'classnames';

import { HelperText } from '../VirtualHearings/HelperText';
import { ReadOnly } from '../details/ReadOnly';
import { enablePadding } from '../details/style';
import COPY from '../../../../COPY';
import TextField from '../../../components/TextField';

export const HearingEmail = ({
  email,
  emailType,
  label,
  readOnly,
  error,
  update,
  required,
  disabled,
  optional,
  helperLabel,
  showHelper,
}) =>
  readOnly ? (
    <ReadOnly label={label} text={email ?? 'None'} />
  ) : (
    <React.Fragment>
      <TextField
        optional={optional}
        readOnly={disabled}
        errorMessage={error}
        name={label}
        value={email}
        required={!disabled && required}
        strongLabel
        className={[
          classnames('cf-form-textinput', 'cf-inline-field', {
            [enablePadding]: error,
          }),
        ]}
        onChange={(newEmail) =>
          update('hearing', {
            [emailType]: isEmpty(newEmail) ? null : newEmail,
          })
        }
      />
      {showHelper ? <HelperText label={helperLabel} /> : null}
    </React.Fragment>
  );

HearingEmail.defaultProps = {
  helperLabel: COPY.VIRTUAL_HEARING_EMAIL_HELPER_TEXT,
  showHelper: true,
};

HearingEmail.propTypes = {
  email: PropTypes.string,
  emailType: PropTypes.string,
  label: PropTypes.string,
  readOnly: PropTypes.bool,
  error: PropTypes.string,
  update: PropTypes.func,
  required: PropTypes.bool,
  optional: PropTypes.bool,
  disabled: PropTypes.bool,
  helperLabel: PropTypes.string,
  showHelper: PropTypes.bool,
};
