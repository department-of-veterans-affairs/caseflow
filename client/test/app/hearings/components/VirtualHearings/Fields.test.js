import React from 'react';
import { mount } from 'enzyme';

import { EmailNotificationFields } from 'app/hearings/components/details/EmailNotificationFields';
import { defaultHearing as hearing } from 'test/data/hearings';
import HEARING_TIME_OPTIONS from 'constants/HEARING_TIME_OPTIONS';
import { Timezone } from 'app/hearings/components/VirtualHearings/Timezone';
import { HearingEmail } from 'app/hearings/components/details/HearingEmail';

describe('Fields', () => {
  const expectations = (fields) => {
    // Emails
    expect(fields.find(HearingEmail)).toHaveLength(2);
    expect(fields.find(HearingEmail).first().
      prop('email')).toEqual(hearing.appellantEmailAddress);
    expect(fields.find(HearingEmail).first().
      prop('label')).toEqual('Veteran Email');
    expect(fields.find(HearingEmail).at(1).
      prop('email')).toEqual(hearing.representativeEmailAddress);

    // Timezones
    expect(fields.find(Timezone)).toHaveLength(2);
    expect(fields.find(Timezone).first().
      prop('value')).toEqual(hearing.appellantTz);
    expect(fields.find(Timezone).at(1).
      prop('value')).toEqual(hearing.representativeTz);

    // Other components
    expect(fields.find('.cf-help-divider')).toHaveLength(1);
    expect(fields).toMatchSnapshot();
  };

  test('Display timezone and divider for Central', () => {
    // Run the test
    const fields = mount(
      <EmailNotificationFields
        appellantTitle="Veteran"
        time={HEARING_TIME_OPTIONS[0].value}
        requestType="Central"
        hearing={hearing}
      />
    );

    expectations(fields);
  });

  test('Display timezone and divider for Video', () => {
    // Run the test
    const fields = mount(
      <EmailNotificationFields
        appellantTitle="Veteran"
        time={HEARING_TIME_OPTIONS[0].value}
        requestType="Video"
        hearing={hearing}
      />
    );

    expectations(fields);
  });
})
;
