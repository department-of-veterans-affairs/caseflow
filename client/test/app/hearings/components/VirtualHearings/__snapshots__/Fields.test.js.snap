// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fields Display timezone and divider for Central 1`] = `
<div>
  <div
    class="cf-app-segment"
  >
    <h2
      data-css-3rhuy2=""
    />
    <div
      data-css-1571z6o=""
    >
      <div
        class="usa-grid"
        id="email-section"
      >
        <div
          class="usa-width-one-third"
          data-css-150oin1=""
        >
          <div
            class=""
          >
            <div
              class="cf-form-dropdown dropdown-appellantTz"
              data-css-1dzedb9=""
            >
              <label
                class="question-label"
                for="appellant-tz"
                id="appellant-tz-label"
              >
                <strong>
                  <span>
                    Veteran Timezone
                    <span
                      class="cf-optional"
                    >
                      Optional
                    </span>
                  </span>
                </strong>
              </label>
              <div
                class="cf-select"
              >
                <div
                  class=" css-2b097c-container"
                >
                  <div
                    class="cf-select__control css-yk16xz-control"
                  >
                    <div
                      class="cf-select__value-container cf-select__value-container--has-value css-g1d714-ValueContainer"
                    >
                      <div
                        class="cf-select__single-value css-1uccc91-singleValue"
                      >
                        Pacific Time (US & Canada) (5:15 AM)
                      </div>
                      <div
                        class="css-81nyic-Input"
                      >
                        <div
                          class="cf-select__input"
                          style="display: inline-block;"
                        >
                          <input
                            aria-autocomplete="list"
                            aria-expanded="false"
                            aria-haspopup="true"
                            aria-label="Pacific Time (US & Canada) (5:15 AM)"
                            aria-labelledby="appellant-tz-label"
                            aria-owns="appellant-tz-listbox"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            id="appellant-tz"
                            role="combobox"
                            spellcheck="false"
                            style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                          <div
                            style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                          />
                        </div>
                      </div>
                    </div>
                    <div
                      class="cf-select__indicators css-1hb7zxy-IndicatorsContainer"
                    >
                      <span
                        class="cf-select__indicator-separator css-1okebmr-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class="cf-select__indicator cf-select__dropdown-indicator css-tlfecz-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-6q0nyr-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <input
                    name="appellantTz"
                    type="hidden"
                    value="America/Los_Angeles"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="helper-text"
          >
            The hearing time in email notifications is calculated based on the timezone.
          </div>
        </div>
        <div
          class="usa-width-one-third"
        >
          <div
            class="cf-form-textinput cf-inline-field   "
          >
            <label
              for="Veteran Email"
            >
              <strong>
                <span>
                  Veteran Email
                  <span
                    class="cf-optional"
                  >
                    Optional
                  </span>
                </span>
              </strong>
            </label>
            <div
              class="input-container"
            >
              <input
                aria-readonly="false"
                class="cf-form-textinput cf-inline-field"
                id="Veteran Email"
                name="Veteran Email"
                type="text"
                value="tom.brady@caseflow.gov"
              />
            </div>
          </div>
          <div
            class="helper-text"
          >
            This email address will be used to send notifications for this hearing only.
          </div>
        </div>
      </div>
      <div
        class="cf-help-divider"
      />
      <div
        class="usa-grid css-1jfbwxw"
      >
        <div
          class="usa-width-one-third"
          data-css-150oin1=""
        >
          <div
            class=""
          >
            <div
              class="cf-form-dropdown dropdown-representativeTz"
              data-css-1dzedb9=""
            >
              <label
                class="question-label"
                for="representative-tz"
                id="representative-tz-label"
              >
                <strong>
                  <span>
                    POA/Representative Timezone
                    <span
                      class="cf-required"
                    >
                      Required
                    </span>
                  </span>
                </strong>
              </label>
              <div
                class="cf-select"
              >
                <div
                  class=" css-2b097c-container"
                >
                  <div
                    class="cf-select__control css-yk16xz-control"
                  >
                    <div
                      class="cf-select__value-container cf-select__value-container--has-value css-g1d714-ValueContainer"
                    >
                      <div
                        class="cf-select__single-value css-1uccc91-singleValue"
                      >
                        Pacific Time (US & Canada) (5:15 AM)
                      </div>
                      <div
                        class="css-81nyic-Input"
                      >
                        <div
                          class="cf-select__input"
                          style="display: inline-block;"
                        >
                          <input
                            aria-autocomplete="list"
                            aria-expanded="false"
                            aria-haspopup="true"
                            aria-label="Pacific Time (US & Canada) (5:15 AM)"
                            aria-labelledby="representative-tz-label"
                            aria-owns="representative-tz-listbox"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            id="representative-tz"
                            role="combobox"
                            spellcheck="false"
                            style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                          <div
                            style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                          />
                        </div>
                      </div>
                    </div>
                    <div
                      class="cf-select__indicators css-1hb7zxy-IndicatorsContainer"
                    >
                      <span
                        class="cf-select__indicator-separator css-1okebmr-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class="cf-select__indicator cf-select__dropdown-indicator css-tlfecz-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-6q0nyr-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <input
                    name="representativeTz"
                    type="hidden"
                    value="America/Los_Angeles"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="helper-text"
          >
            The hearing time in email notifications is calculated based on the timezone.
          </div>
        </div>
        <div
          class="usa-width-one-third"
        >
          <div
            class="cf-form-textinput cf-inline-field   "
          >
            <label
              for="POA/Representative Email"
            >
              <strong>
                <span>
                  POA/Representative Email
                  <span
                    class="cf-optional"
                  >
                    Optional
                  </span>
                </span>
              </strong>
            </label>
            <div
              class="input-container"
            >
              <input
                aria-readonly="false"
                class="cf-form-textinput cf-inline-field"
                id="POA/Representative Email"
                name="POA/Representative Email"
                type="text"
                value="tom.brady@caseflow.gov"
              />
            </div>
          </div>
          <div
            class="helper-text"
          >
            This email address will be used to send notifications for this hearing only.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Fields Display timezone and divider for Video 1`] = `
<div>
  <div
    class="cf-app-segment"
  >
    <h2
      data-css-3rhuy2=""
    />
    <div
      data-css-1571z6o=""
    >
      <div
        class="usa-grid"
        id="email-section"
      >
        <div
          class="usa-width-one-third"
          data-css-150oin1=""
        >
          <div
            class=""
          >
            <div
              class="cf-form-dropdown dropdown-appellantTz"
              data-css-1dzedb9=""
            >
              <label
                class="question-label"
                for="appellant-tz"
                id="appellant-tz-label"
              >
                <strong>
                  <span>
                    Veteran Timezone
                    <span
                      class="cf-optional"
                    >
                      Optional
                    </span>
                  </span>
                </strong>
              </label>
              <div
                class="cf-select"
              >
                <div
                  class=" css-2b097c-container"
                >
                  <div
                    class="cf-select__control css-yk16xz-control"
                  >
                    <div
                      class="cf-select__value-container cf-select__value-container--has-value css-g1d714-ValueContainer"
                    >
                      <div
                        class="cf-select__single-value css-1uccc91-singleValue"
                      >
                        Pacific Time (US & Canada) (5:15 AM)
                      </div>
                      <div
                        class="css-81nyic-Input"
                      >
                        <div
                          class="cf-select__input"
                          style="display: inline-block;"
                        >
                          <input
                            aria-autocomplete="list"
                            aria-expanded="false"
                            aria-haspopup="true"
                            aria-label="Pacific Time (US & Canada) (5:15 AM)"
                            aria-labelledby="appellant-tz-label"
                            aria-owns="appellant-tz-listbox"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            id="appellant-tz"
                            role="combobox"
                            spellcheck="false"
                            style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                          <div
                            style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                          />
                        </div>
                      </div>
                    </div>
                    <div
                      class="cf-select__indicators css-1hb7zxy-IndicatorsContainer"
                    >
                      <span
                        class="cf-select__indicator-separator css-1okebmr-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class="cf-select__indicator cf-select__dropdown-indicator css-tlfecz-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-6q0nyr-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <input
                    name="appellantTz"
                    type="hidden"
                    value="America/Los_Angeles"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="helper-text"
          >
            The hearing time in email notifications is calculated based on the timezone.
          </div>
        </div>
        <div
          class="usa-width-one-third"
        >
          <div
            class="cf-form-textinput cf-inline-field   "
          >
            <label
              for="Veteran Email"
            >
              <strong>
                <span>
                  Veteran Email
                  <span
                    class="cf-optional"
                  >
                    Optional
                  </span>
                </span>
              </strong>
            </label>
            <div
              class="input-container"
            >
              <input
                aria-readonly="false"
                class="cf-form-textinput cf-inline-field"
                id="Veteran Email"
                name="Veteran Email"
                type="text"
                value="tom.brady@caseflow.gov"
              />
            </div>
          </div>
          <div
            class="helper-text"
          >
            This email address will be used to send notifications for this hearing only.
          </div>
        </div>
      </div>
      <div
        class="cf-help-divider"
      />
      <div
        class="usa-grid css-1jfbwxw"
      >
        <div
          class="usa-width-one-third"
          data-css-150oin1=""
        >
          <div
            class=""
          >
            <div
              class="cf-form-dropdown dropdown-representativeTz"
              data-css-1dzedb9=""
            >
              <label
                class="question-label"
                for="representative-tz"
                id="representative-tz-label"
              >
                <strong>
                  <span>
                    POA/Representative Timezone
                    <span
                      class="cf-required"
                    >
                      Required
                    </span>
                  </span>
                </strong>
              </label>
              <div
                class="cf-select"
              >
                <div
                  class=" css-2b097c-container"
                >
                  <div
                    class="cf-select__control css-yk16xz-control"
                  >
                    <div
                      class="cf-select__value-container cf-select__value-container--has-value css-g1d714-ValueContainer"
                    >
                      <div
                        class="cf-select__single-value css-1uccc91-singleValue"
                      >
                        Pacific Time (US & Canada) (5:15 AM)
                      </div>
                      <div
                        class="css-81nyic-Input"
                      >
                        <div
                          class="cf-select__input"
                          style="display: inline-block;"
                        >
                          <input
                            aria-autocomplete="list"
                            aria-expanded="false"
                            aria-haspopup="true"
                            aria-label="Pacific Time (US & Canada) (5:15 AM)"
                            aria-labelledby="representative-tz-label"
                            aria-owns="representative-tz-listbox"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            id="representative-tz"
                            role="combobox"
                            spellcheck="false"
                            style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                          <div
                            style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                          />
                        </div>
                      </div>
                    </div>
                    <div
                      class="cf-select__indicators css-1hb7zxy-IndicatorsContainer"
                    >
                      <span
                        class="cf-select__indicator-separator css-1okebmr-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class="cf-select__indicator cf-select__dropdown-indicator css-tlfecz-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-6q0nyr-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <input
                    name="representativeTz"
                    type="hidden"
                    value="America/Los_Angeles"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="helper-text"
          >
            The hearing time in email notifications is calculated based on the timezone.
          </div>
        </div>
        <div
          class="usa-width-one-third"
        >
          <div
            class="cf-form-textinput cf-inline-field   "
          >
            <label
              for="POA/Representative Email"
            >
              <strong>
                <span>
                  POA/Representative Email
                  <span
                    class="cf-optional"
                  >
                    Optional
                  </span>
                </span>
              </strong>
            </label>
            <div
              class="input-container"
            >
              <input
                aria-readonly="false"
                class="cf-form-textinput cf-inline-field"
                id="POA/Representative Email"
                name="POA/Representative Email"
                type="text"
                value="tom.brady@caseflow.gov"
              />
            </div>
          </div>
          <div
            class="helper-text"
          >
            This email address will be used to send notifications for this hearing only.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
